WarpBuf/src/WarpBufChugin.cpp

```diff
@@ -451,40 +149,44 @@ double WarpBufChugin::getTranspose() {
     return transpose;
 }
 
-void WarpBufChugin::setTranspose(double transpose) {
+void
+WarpBufChugin::setTranspose(double transpose) {
 
-    float scale = std::pow(2., transpose/12.);
+    double scale = std::pow(2., transpose/12.);
 
     m_rbstretcher->setPitchScale(scale);
 }
 
-double WarpBufChugin::getBPM() {
+double
+WarpBufChugin::getBPM() {
     return m_bpm;
 }
 
-void WarpBufChugin::setBPM(double bpm) {
+void
+WarpBufChugin::setBPM(double bpm) {
     if (bpm <= 0) {
         std::cerr << "Error: BPM must be positive." << std::endl;
         return;
     }
     m_bpm = bpm;
 }
 
-void WarpBufChugin::tick(SAMPLE* in, SAMPLE* out, int nframes)
+void
+WarpBufChugin::tick(SAMPLE* in, SAMPLE* out, int nframes)
 {
+    allocate(sfinfo.channels, nframes);
+
     bool past_end_marker_and_loop_off = m_playHeadBeats > m_clipInfo.end_marker && !m_clipInfo.loop_on;
-    if (past_end_marker_and_loop_off || (!m_play) || (!m_fileWasRead)) {
+    if (m_channels==0 || sfinfo.channels == 0 || past_end_marker_and_loop_off || (!m_play)) {
```

Comment:
Could you add a comment about this behavior? I think I understand why the conditions are there but not 100% sure.